src/otx/core/model/entity/segmentation.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Class definition for detection model entity used in OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.segmentation import SegBatchDataEntity, SegBatchPredEntity
+from otx.core.model.entity.base import OTXModel
+from otx.core.utils.build import build_mm_model
+
+if TYPE_CHECKING:
+    from mmseg.models.data_preprocessor import SegDataPreProcessor
+    from omegaconf import DictConfig
+    from torch import nn
+
+
+class OTXSegmentationModel(OTXModel[SegBatchDataEntity, SegBatchPredEntity]):
+    """Base class for the detection models used in OTX."""
+
+# This is an example for MMDetection models
+# In this way, we can easily import some models developed from the MM community
+class MMSegCompatibleModel(OTXSegmentationModel):
+    """Segmentation model compatible for MMSeg.
+
+    It can consume MMSeg model configuration translated into OTX configuration
+    (please see otx.tools.translate_mmrecipe) and create the OTX detection model
+    compatible for OTX pipelines.
+    """
+
+    def __init__(self, config: DictConfig) -> None:
+        self.config = config
+        self.load_from = self.config.pop("load_from", None)
+        super().__init__()
+
+    def _create_model(self) -> nn.Module:
+        from mmengine.registry import MODELS as MMENGINE_MODELS
+        from mmseg.registry import MODELS
+
+        seg = MODELS.get("SegDataPreProcessor")
+        MMENGINE_MODELS.register_module(module=seg)
+
+        if self.config.backbone.type == 'LiteHRNet':
+            model = build_mm_model(self.config, MODELS, self.load_from)
+        else:
+            # for other mmseg models load only backbone
+            model = build_mm_model(self.config, MODELS, self.load_from, True)
```

Comment:
I don't want to see anymore the hard-coded `if-else` statements like OTX v1, which is hidden at the runtime, so that hardly to track. Please consider more explicit way to solve this problem. For example, you can add `load_backbone_only` configuration variable to the `LiteHRNet` templates's `model.otx_model.config`.

Reply:
Agree, I will try to avoid this

Reply:
I removed this, we can use init_cfg and model.init_weights() actually to load backbone only